Sources/Search.php

```diff
@@ -688,14 +688,34 @@ function PlushSearch2()
 		$stripped_query = strtr($stripped_query, array('"' => ''));
 
 	$no_regexp = preg_match('~&#(?:\d{1,7}|x[0-9a-fA-F]{1,6});~', $stripped_query) === 1;
+	$is_search_regex = !empty($modSettings['search_match_words']) || !$no_regexp;
+
+	// Specify the function to search with. Regex is for word boundaries.
+	$regex_keyword = $smcFunc['db_title'] === POSTGRE_TITLE ? '~*' : 'REGEXP';
+	$query_match_type = $is_search_regex ? $regex_keyword : 'LIKE';
+	db_extend();
+	$version = $smcFunc['db_get_version']();
+	$supports_pcre = $smcFunc['db_title'] === POSTGRE_TITLE || version_compare($version, strpos($version, 'MariaDB') !== false ? '10.0.5' : '8.0.4', '>=');
```

Comment:
This logic should be in the db layer:
https://github.com/SimpleMachines/SMF/blob/release-2.1/Sources/DbSearch-mysql.php
https://github.com/SimpleMachines/SMF/blob/release-2.1/Sources/DbSearch-postgresql.php

Reply:
and done